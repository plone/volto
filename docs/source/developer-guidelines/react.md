---
myst:
  html_meta:
    "description": "TBD"
    "property=og:description": "TBD"
    "property=og:title": "React"
    "keywords": "Volto, Plone, frontend, React"
---

# React

```{note}
This documentation is a work in progress. Any help is welcome to fill in the
gaps!
```
