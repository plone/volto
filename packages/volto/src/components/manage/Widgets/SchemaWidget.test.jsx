import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-intl-redux';

import SchemaWidget from './SchemaWidget';

jest.mock('@plone/volto/helpers/Loadable/Loadable');
beforeAll(
  async () =>
    await require('@plone/volto/helpers/Loadable/Loadable').__setLoadables(),
);

const mockStore = configureStore();

test('renders a schema widget component', () => {
  const store = mockStore({
    intl: {
      locale: 'en',
      messages: {},
    },
  });

  const { container } = render(
    <Provider store={store}>
      <SchemaWidget
        id="my-field"
        title="My field"
        onChange={() => {}}
        value={'{"fieldsets": [{"fields": []}]}'}
      />
    </Provider>,
  );

  expect(container).toMatchSnapshot();
});
