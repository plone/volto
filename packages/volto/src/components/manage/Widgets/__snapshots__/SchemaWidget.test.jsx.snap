// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a schema widget component 1`] = `
<div>
  <div
    className="ui segments"
    style={
      Object {
        "margin": "-1rem",
      }
    }
  >
    <div
      className="ui pointing secondary attached tabular menu"
      role="tablist"
      style={
        Object {
          "background": "transparent",
          "display": "flex",
          "flexDirection": "row",
          "flexWrap": "wrap",
        }
      }
    >
      <div
        className="item active"
        data-rbd-draggable-context-id="0"
        onClick={[Function]}
        onTransitionEnd={null}
        style={
          Object {
            "background": "transparent",
            "transform": null,
            "transition": null,
            "userSelect": "none",
          }
        }
      >
        <i
          aria-describedby="rbd-hidden-text-0-hidden-text-0"
          aria-hidden="true"
          className="grey bars icon drag handle"
          data-rbd-drag-handle-context-id="0"
          draggable={false}
          onDragStart={[Function]}
          role="button"
          tabIndex={0}
        />
        <button
          className="item ui noborder button"
          onClick={[Function]}
        >
          <i
            aria-hidden="true"
            className="blue write square large icon"
            onClick={[Function]}
          />
        </button>
        <button
          className="item ui noborder button"
          onClick={[Function]}
        >
          <i
            aria-hidden="true"
            className="red close large icon"
            onClick={[Function]}
          />
        </button>
      </div>
      <div
        className="item item-add"
      >
        <button
          aria-label="Add"
          className="item ui noborder button"
          onClick={[Function]}
        >
          <i
            aria-hidden="true"
            className="plus large icon"
            onClick={[Function]}
          />
        </button>
      </div>
    </div>
    <div
      style={
        Object {
          "background": "transparent",
        }
      }
    />
    <div
      className="inline field"
    >
      <div
        className="ui grid"
      >
        <div
          className="stretched row"
        >
          <div
            className="twelve wide column"
          >
            <div
              className="wrapper"
            >
              <label
                htmlFor="addfield"
              >
                Add field
              </label>
            </div>
            <div
              className="toolbar"
            >
              <button
                aria-label="Add"
                className="item ui noborder button"
                id="addfield"
                onClick={[Function]}
              >
                <i
                  aria-hidden="true"
                  className="blue plus large icon"
                  onClick={[Function]}
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
